hibernate5/src/main/java/com/fasterxml/jackson/datatype/hibernate5/HibernateProxySerializer.java

```diff
@@ -182,8 +191,28 @@ protected Object findProxied(HibernateProxy proxy)
                 if (_mapping != null) {
                     idName = _mapping.getIdentifierPropertyName(init.getEntityName());
                 } else {
-                    final SessionImplementor session = init.getSession();
-                    if (session != null) {
+                    // Detect abi compatibility if not set yet
+                    if(hibernateAbiIncompatible == null){
```

Comment:
I think this should be done just once from constructor or even static block: calling it every time serialization occurs is going to be quite inefficient.